### PR TITLE
Dm 1189 cppcheck

### PR DESCRIPTION
**Pull request by SkyToGround**
### Description of work

My initial intent was originally to split this pull request into two. However, seeing as the number of lines of code added is only 355 and that most of the changes are very simple, I decided to create only one pull request for now. Please tell me if you would like me to split the pull request.

This is what I have done:
* Updated cppcheck to the latest version (1.86).
* Fixed a lot of issues found by cppcheck.
* Removed a bunch of dead code related to issues found by cppcheck.
* Changed naming to follow the LLVM standard in code that I touched anyway.

What I have **NOT** done:
* Fix issues that I consider false positives.
* Convert loops into stl algorithm calls, as suggested by the latest version of cppcheck. I might do this at some later point though. 
* Remove dead code not pointed out by cppcheck.

We are now down to [26 issues](https://jenkins.esss.dk/dm/job/ess-dmsc/job/kafka-to-nexus/job/DM-1189_cppcheck_base/1/warnings2Result/) (from 131) of which around half are false positives and the rest are suggestions to replace loops with stl algorithm calls.

**NOTE**
The file _src/schemas/hs00/WriterTyped.h_ contains (almost) the only changes that I would consider non-trivial. The issues found were several uses of shadow variables. Furthermore, the file has several [long methods](https://refactoring.guru/smells/long-method), no unit test coverage and (as far as I can determine) no system/integration tests. I will leave it to you @dominikwerder to determine if you need to run manual tests of this code.

### Issues

This pull request relates to DM-1189 but does not close the issue as more pull requests are coming.

### Acceptance Criteria

Examine the code and determine of it appears sensible and that it introduces no changes in functionality.

### Unit Tests

No unit tests were added or removed. However, several were modified to fix issues found by cppcheck.

### Other

N/A

---

## Code Review (To be filled in by the reviewer only)

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel (does someone want to automate this?).

### CODE REVIEW

File: src/schemas/hs00/Exceptions.cpp [hunk 1]
```diff
@@ -2,11 +2,6 @@
 
 namespace FileWriter {
 namespace Schemas {
-namespace hs00 {
-
-std::runtime_error unimplemented() {
-  return std::runtime_error("unimplemented");
-}
-}
+namespace hs00 {}
```
[the remark this answers is not in the document]
Comment (SkyToGround) [reply src/schemas/hs00/Exceptions.cpp]: Hm, it does not. I did not think of removing it but I will do that now.

**Comment by matthew-d-jones** [comment]
@SkyToGround I have a couple of questions about this PR:
- Is `amo0` not in use at SINQ?
- Why are there tests commented-out?

**Comment by SkyToGround** [comment]
@matthew-d-jones I am assuming that you are thinking of the test on line 1115 in `src/tests/HDFFile.cpp`? It was disabled by @dominikwerder in March of last year. I suggest that you ask him about the reason for it. I commented out the function that the test called to get rid of a warning from cppcheck about unused code. Seeing as I did not want to change any functionality, this was the best option in my opinion but I am happy hear if you have suggestions.

**Comment by matthew-d-jones** [comment]
@michele-brambilla @mkoennecke @dominikwerder Is the `amo0` schema in use? It doesn't seem to be used on `master` branch.

**Comment by matthew-d-jones** [comment]
@SkyToGround Ah, I think the test was disabled due to an issue in `h5cpp`. If we merge the PR where I've updated the version of `h5cpp` (#382) then we can merge master in here and see if we can re-enable the test now.

**Comment by SkyToGround** [comment]
@matthew-d-jones Then maybe I suggest that you create a branch where you update h5cpp and try to re-enable the test? Alternatively, we can do that after this branch has been merged? It feels wrong to go around changing functionality in this pull-request.

**Comment by matthew-d-jones** [comment]
@SkyToGround Sure, I've created JIRA ticket https://jira.esss.lu.se/browse/DM-1248 to look into whether the test can be re-enabled.

**Pull request merged**